### ISSUE TITLE
Wildcard V2: Implement `<Checkbox />` codemod

### ISSUE THREAD

**Issue by valerybugakov** [sourcegraph/sourcegraph#33267]
## Problem statement

Manual migration is required to update our code to use the `<Checkbox />` Wildcard component.
Check out [the parent issue](https://github.com/sourcegraph/sourcegraph/issues/27719) for more details.

## Success criteria

A new `Checkbox` codemod with test cases is added to [the codemod repo](https://github.com/sourcegraph/codemod).

## Implementation details

Use [the `Input` codemod](https://github.com/sourcegraph/codemod/blob/main/packages/transforms/src/inputToComponent/inputToComponent.ts) as an example.

## Time estimate

- This issue should take **4 hours** at maximum. Ping the reviewer in the spec pull request if time-consuming changes are required.

**Comment by gitstart-sourcegraph** [comment]
@valerybugakov @umpox 
While implementing this tool, we noticed `Checkbox` wildcard component has a required prop `label`, however most use cases uses such layout as:
```
<div className="form-check">
<label className="form-check-label">
    <input
            type="checkbox"
            className="form-check-input"
            {...otherProps}
     />
     {label}
</label>
</div>
```
i.e we would need to handle such use cases in the codemod tool `div.form-check > input[type=checkbox] + label.form-check-label`, to ensure we can pass in the parent `label` as a Prop into the wildcard's `Checkbox` component.
WDYT about making codemod tool to support this use case? 

BTW this would require more time than initial estimate / manual migration.

**Comment by valerybugakov** [comment]
@gitstart-sourcegraph, thanks for raising this question! We reviewed the `Checkbox` component API and will update it first to prepare for the migration. We also noticed that there're 41 occurrences of `type="checkbox" in the codebase, so it makes sense to migrate them manually to avoid building codemods for each layout. 

@umpox will look into the `Checkbox` API today, and after that, we can start working on the manual migration issue [here](https://github.com/sourcegraph/sourcegraph/issues/27719).
